mlflow/sklearn/__init__.py

```diff
@@ -455,6 +457,60 @@ def load_model(model_uri):
     )
 
 
+class _AutologTrainingStatus:
+    def __init__(self):
+        self.model_id_to_run_id_map = {}
+        self.dataset_id_to_dataset_name_and_model_id = {}
+        self.in_fit_call_scope = False
+        self.in_scorer_call_scope = False
+        self.in_eval_and_log_metrics_scope = False
+        self.in_model_score_call_scope = False
```

Comment:
Can we use a single piece of state - `should_log_eval_metrics = True` - and have each environment disable eval metrics using this boolean? I don't think we need a separate piece of boolean state for each of these circumstances. Please let me know if I'm missing something.

Can we also create a context manager to use instead of try/finally? Something like

```
with disable_log_eval_metrics():
    # call model.score(), etc
```

Reply:
A context manager would also allow us to remove these `global ...` statements from several parts of the implementation.